torch/nn/modules/activation.py

```diff
@@ -1113,7 +1113,7 @@ def forward(self, query: Tensor, key: Tensor, value: Tensor, key_padding_mask: O
                 why_not_fast_path = "some Tensor argument has_torch_function"
             elif not all([(x is None or x.is_cuda or 'cpu' in str(x.device)) for x in tensor_args]):
                 why_not_fast_path = "some Tensor argument is neither CUDA nor CPU"
-            elif torch.is_grad_enabled() and any([x.requires_grad for x in tensor_args]):
+            elif torch.is_grad_enabled() and any([x is not None and x.requires_grad for x in tensor_args]):
```

Comment:
```suggestion
            elif torch.is_grad_enabled() and any((x is not None and x.requires_grad for x in tensor_args))
```
is more efficient and allows for shortcircuiting (while the list comprehension version does not). This should probably be applied all over this file though.